### PR TITLE
[Torch] add onnx2tensorrt backend

### PR DESCRIPTION
**Pull request by fortianyou**

### CODE REVIEW

**Comment by Orion34-lanbo** [comment]
Both bazel BUILD files and cmake files exist，how these two build system work together?
Plus, some of the design docs are still in Chinese...

**Review by qiuxiafei — COMMENTED** [review]
Also, maybe we could change *MLIR* in variable/namespace/method names to *DISC* ...

File: pytorch_blade/src/compiler/mlir/runtime/disc_engine.cpp [hunk 1]
```diff
@@ -57,6 +91,10 @@ std::shared_ptr<DiscEngine> DiscEngine::Create(const State& engine_state) {
   return std::shared_ptr<DiscEngine>(new DiscEngine(engine_state));
 }
 
+const char* GetBackendName() {
```
[the remark this answers is not in the document]
Comment (fortianyou) [reply pytorch_blade/src/compiler/mlir/runtime/disc_engine.cpp:94]: It's sad that the declaration of TRTEngine in the header files would pollute other files which include them. Such as the dependency on `cuda_runtime_api.h` will be forwarded to `tensorrt/pybind_functions.cpp`.

To make the dependencies more relaxed I had to move the whole class implementations of DiscEngine and TRTEngine into the c++ sources. Unfortunately， we had to trade off the code readability.

**Review by linearhit — APPROVED** [review]
LGTM，will the documents be updated in a separate CR?

**Comment by fortianyou** [comment]
> LGTM，will the documents be updated in a separate CR?

Yes, I will commit the documents later.

**Pull request merged**